internal/plugin/grpc_client_plugin.go

```diff
@@ -10,64 +10,79 @@ import (
 	"fmt"
 	"log/slog"
 	"net"
-	"time"
 
 	"github.com/google/uuid"
-	grpcRetry "github.com/grpc-ecosystem/go-grpc-middleware/retry"
 	"github.com/nginx/agent/v3/api/grpc/mpi/v1"
+	"github.com/nginx/agent/v3/internal/backoff"
 	"github.com/nginx/agent/v3/internal/bus"
 	"github.com/nginx/agent/v3/internal/config"
+	agentGrpc "github.com/nginx/agent/v3/internal/grpc"
 	"google.golang.org/grpc"
-	"google.golang.org/grpc/credentials/insecure"
-	"google.golang.org/grpc/keepalive"
+	"google.golang.org/grpc/connectivity"
 	"google.golang.org/protobuf/types/known/timestamppb"
 
 	uuidLibrary "github.com/nginx/agent/v3/internal/uuid"
 )
 
-const (
-	DefaultClientTime   = 120  // Default time for client operations in seconds
-	DefaultTimeout      = 60   // Default timeout duration in seconds
-	DefaultPermitStream = true // Flag indicating permission of stream
-)
-
 type (
 	GrpcClient struct {
 		messagePipe bus.MessagePipeInterface
 		config      *config.Config
 		conn        *grpc.ClientConn
 		cancel      context.CancelFunc
+		settings    *backoff.Settings
 	}
 )
 
 func NewGrpcClient(agentConfig *config.Config) *GrpcClient {
 	if agentConfig != nil && agentConfig.Command.Server.Type == "grpc" {
+		if agentConfig.Common == nil {
+			slog.Error("invalid configuration settings")
+			return nil
+		}
+		settings := &backoff.Settings{
+			InitialInterval: agentConfig.Common.InitialInterval,
+			MaxInterval:     agentConfig.Common.MaxInterval,
+			MaxElapsedTime:  agentConfig.Common.MaxElapsedTime,
+			Jitter:          agentConfig.Common.Jitter,
+			Multiplier:      agentConfig.Common.Multiplier,
+		}
+
 		return &GrpcClient{
-			config: agentConfig,
+			config:   agentConfig,
+			settings: settings,
 		}
 	}
 
 	return nil
 }
 
 func (gc *GrpcClient) Init(ctx context.Context, messagePipe bus.MessagePipeInterface) error {
-	slog.Debug("Starting grpc client plugin")
+	slog.Info("Starting grpc client plugin")
 	gc.messagePipe = messagePipe
+	var grpcClientCtx context.Context
+	var err error
 
 	serverAddr := net.JoinHostPort(
 		gc.config.Command.Server.Host,
 		fmt.Sprint(gc.config.Command.Server.Port),
 	)
 
-	var grpcClientCtx context.Context
 	grpcClientCtx, gc.cancel = context.WithTimeout(ctx, gc.config.Client.Timeout)
+	slog.Info("Dialing grpc server", "server_addr", serverAddr)
 
-	var err error
-	gc.conn, err = grpc.DialContext(grpcClientCtx, serverAddr, gc.getDialOptions()...)
+	gc.conn, err = grpc.DialContext(grpcClientCtx, serverAddr, agentGrpc.GetDialOptions(gc.config)...)
 	if err != nil {
 		return err
 	}
+	backOffCtx, backoffCancel := context.WithTimeout(ctx, gc.config.Client.Timeout)
```

Comment:
Just wanna confirm that `gc.config.Client.Timeout` is the correct timeout here? This means that `gc.config.Client.Timeout` should always be greater or equal to `gc.settings.MaxElapsedTime`.